verilog/tools/ls/symbol-table-handler.cc

```diff
@@ -0,0 +1,41 @@
+#include "verilog/tools/ls/symbol-table-handler.h"
+
+namespace verilog {
+
+void SymbolTableHandler::setProject(
+    absl::string_view root, const std::vector<std::string> &include_paths,
+    absl::string_view corpus) {
+  currproject = std::make_shared<VerilogProject>(root, include_paths, corpus);
+  resetSymbolTable();
+}
+
+void SymbolTableHandler::resetSymbolTable() {
+  checkedfiles.clear();
+  symboltable = std::make_shared<SymbolTable>(currproject.get());
+}
+
+void SymbolTableHandler::buildSymbolTableFor(VerilogSourceFile &file) {
+  auto result = BuildSymbolTable(file, symboltable.get(), currproject.get());
+}
+
+std::vector<verible::lsp::Location> SymbolTableHandler::findDefinition(
+    const verible::lsp::DefinitionParams &params) {
+  auto filepath = params.textDocument.uri;
+  if (checkedfiles.find(filepath) == checkedfiles.end()) {
+    // File hasn't been tracked yet in the symbol table, add it
+    auto openedfile = currproject->OpenTranslationUnit(filepath);
```

Comment:
Roughly here is where you want to open the file _not_ from the filesystem, but get it from the BufferTracker.
There you should alerady get some pre-parsed content.

For this use case it is probably sufficient to request the `BufferTracker::last_good()` buffer.

Reply:
Yes, reading the files from VerilogProject was the initial step, since the base implementation of the SymbolTable relies on VerilogProject and its source containers.

In the next stage, the idea is to use existing parsed/raw buffers provided by the language server, and fallback to actual filesystem paths when necessary.

Reply:
The buffers from LS client are used by default (last_good for symbol table updating)